addons/godot-next/nodes/behaviors.gd

```diff
@@ -0,0 +1,173 @@
+# author: xdgamestudios
+# license: MIT
+# description: Provides a behavior system API for the nodes. 
+# todo: Refactor code in general to clean and improve performance
+# usage:
+# - Creating:
+#     behaviors = Behaviors.new() # through code
+#     Note:
+#     - To use in the inspector, just add node of type Behaviors.
+# - Adding Behaviors:
+#     behaviors.add_behavior(MyBehavior) # returns instance of MyBehavior
+#     Note:
+#     - Allows only one behavior of each type.
+#     - Returns null if the user assigns an invalid Behavior class. Else returns the added or pre-existing Behavior.
```

Comment:
If you have this one comment (which you can place on the same line as the code sample itself), you don't need any of the other notes since it explains all the same information in a more succinct statement.

Reply:
(If you don't want the line to be overly long, then just place the comment on a line above the code sample)